services/explorer-indexer/src/modules/dag/dag.service.ts

```diff
@@ -59,4 +65,15 @@ export default class DagService {
     const updated = await this.dagRepository.save(dag);
     if (updated) this.logger.log(`DAG ${updateData.block} finalized`);
   }
+
+  private async updateNode(dag: DagEntity) {
+    const foundNode = await this.nodeRepository.findOne({
+      where: { address: dag.author },
+    });
+    if (foundNode) {
+      foundNode.dagCount += 1;
```

Comment:
It might be a more robust way of doing this to implement a query that returns the # of dags where the node's address is its author. WDYT @Malak67 ?

Reply:
Yeah, I am also not that happy with this way of doing it. We can discuss it and see what other options we have.